File: UNIDAD 1/01_SOA_CON_BDD/00. SOAP_JAVA_EUREKABANK_GR10/02. CLIESC/EUREKABANK_CLIESC_GR10/src/ec/edu/monster/servicio/EurekaService.java
package ec.edu.monster.servicio;

import ec.edu.monster.ws.Movimiento;
import ec.edu.monster.ws.WSEureka;
import ec.edu.monster.ws.WSEureka_Service;
import java.util.List;

/**
 * Servicio para consumir las operaciones del web service WSEureka.
 *
 * @author MATIAS
 */
public class EurekaService {

    private final WSEureka port;

    public EurekaService() {
        WSEureka_Service service = new WSEureka_Service();
        this.port = service.getWSEurekaPort();
    }

    /**
     * Obtiene la lista de movimientos para una cuenta específica.
     *
     * @param cuenta Código de la cuenta.
     * @return Lista de movimientos.
     */
    public List<Movimiento> traerMovimientos(String cuenta) {
        return port.leerMovimientos(cuenta);
    }

    /**
     * Registra un depósito en una cuenta.
     *
     * @param cuenta Código de la cuenta.
     * @param importe Monto del depósito.
     * @param codEmp Código del empleado.
     * @return 1 si el depósito es exitoso, -1 si falla.
     */
    public int registrarDeposito(String cuenta, double importe, String codEmp) {
        return port.registrarDeposito(cuenta, importe, codEmp);
    }

    /**
     * Registra un retiro en una cuenta.
     *
     * @param cuenta Código de la cuenta.
     * @param importe Monto del retiro.
     * @param codEmp Código del empleado.
     * @return 1 si el retiro es exitoso, -1 si falla.
     */
    public int registrarRetiro(String cuenta, double importe, String codEmp) {
        return port.registrarRetiro(cuenta, importe, codEmp);
    }

    /**
     * Realiza una transferencia entre dos cuentas.
     *
     * @param cuentaOrigen Código de la cuenta origen.
     * @param cuentaDestino Código de la cuenta destino.
     * @param importe Monto de la transferencia.
     * @param codEmp Código del empleado.
     * @return 1 si la transferencia es exitosa, -1 si falla.
     */
    public int realizarTransferencia(String cuentaOrigen, String cuentaDestino, double importe, String codEmp) {
        return port.realizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp);
    }

    /**
     * Verifica el saldo de una cuenta.
     *
     * @param cuenta Código de la cuenta.
     * @return Saldo de la cuenta, o -1 si hay error.
     */
    public double verificarSaldo(String cuenta) {
        return port.verificarSaldo(cuenta);
    }

    /**
     * Obtiene el costo de movimiento para una cuenta.
     *
     * @param cuenta Código de la cuenta.
     * @return Costo del movimiento, o -1 si hay error.
     */
    public double obtenerCostoMovimiento(String cuenta) {
        return port.obtenerCostoMovimiento(cuenta);
    }

    /**
     * Inicia sesión para un usuario.
     *
     * @param usuario Nombre de usuario (vch_emplusuario).
     * @param clave Clave del usuario (vch_emplclave).
     * @return Objeto Usuario si la autenticación es exitosa, null si falla.
     */
    public ec.edu.monster.modelo.Usuario iniciarSesion(String usuario, String clave) {
        ec.edu.monster.ws.Usuario wsUser = port.iniciarSesion(usuario, clave);
        if (wsUser != null) {
            return new ec.edu.monster.modelo.Usuario(
                    wsUser.getCodigo(),
                    wsUser.getUsuario(),
                    wsUser.getClave(),
                    wsUser.getEstado()
            );
        }
        return null;
    }

}

File: UNIDAD 1/01_SOA_CON_BDD/001. SOAP_DOTNET_EUREKABANK_GR10/02. CLICON/EurekaBankConsoleClient/EurekaBankConsoleClient/Servicio/EurekaService.cs
using EurekaBankConsoleClient;
using EurekaServiceReference;
using System.Collections.Generic;

namespace EurekaBankConsoleClient.Servicio
{
    public class EurekaService
    {
        private readonly EurekaBankServiceClient port;

        public EurekaService()
        {
            port = new EurekaBankServiceClient();
        }

        public List<Movimiento> TraerMovimientos(string cuenta)
        {
            return new List<Movimiento>(port.LeerMovimientos(cuenta));
        }

        public int RegistrarDeposito(string cuenta, double importe, string codEmp)
        {
            return port.RegistrarDeposito(cuenta, importe, codEmp);
        }

        public int RegistrarRetiro(string cuenta, double importe, string codEmp)
        {
            return port.RegistrarRetiro(cuenta, importe, codEmp);
        }

        public int RealizarTransferencia(string cuentaOrigen, string cuentaDestino, double importe, string codEmp)
        {
            return port.RealizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp);
        }

        public double VerificarSaldo(string cuenta)
        {
            return port.VerificarSaldo(cuenta);
        }

        public double ObtenerCostoMovimiento(string cuenta)
        {
            return port.ObtenerCostoMovimiento(cuenta);
        }

        public Usuario IniciarSesion(string usuario, string clave)
        {
            return port.IniciarSesion(usuario, clave);
        }
    }
}